tests/algorithms/test_optimizers.py

```diff
@@ -0,0 +1,80 @@
+"""Tests for non-linear optimization algorithms."""
+
+# Copyright 2024 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import pytest
+import torch
+
+from mrpro.algorithms import adam
+from mrpro.algorithms import lbfgs
+from mrpro.operators import ConstraintsOp
+from tests import RandomGenerator
+from tests.operators._OptimizationTestFunctions import Rosenbrock
+
+
+@pytest.mark.parametrize(
+    'optimizer_name, bounds_flag',
+    [
+        ('adam', 0),
```

Comment:
Question: why is this not a bool?
Like contraint_value True or False?
Do bools not work there?

Reply:
done